### ISSUE TITLE
Support graceful-fs file queueing

### ISSUE THREAD

**Issue by thenickcox** [thlorenz/readdirp#26]
This module has been generally great to work with. However, when I recurse through all directories on a large project, I hit my max file descriptor limit (currently 256). Sure, I could raise it, but I'm working on a script that Travis CI will be running, and I can't count on a high file descriptor limit there.

This module depends on `graceful-fs`, which, from the docs, "Queues up `open` and `readdir` calls, and retries them once something closes if there is an `EMFILE` error from too many file descriptors.

Is it possible to add that queuing functionality to the `readdirp` stream to avoid hitting the file descriptor limit?

### Edit

I guess I should provide info about the error.

```shell
# lib/index.js is a node module that operates on all the files from the readdirp stream
Nick-Cox% node lib/index.js
(libuv) kqueue(): Too many open files
(libuv) kqueue(): Too many open files

events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: write EBADF
```

**Comment by thlorenz** [comment]
Is this fixed via fixing #25?

If not what needs to be done here (PRs welcome).

**Comment by thenickcox** [comment]
Seems to be. Thanks!
